security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/ViolationLocationDeserializer.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.security.BusBreakerViolationLocation;
+import com.powsybl.security.NodeBreakerViolationLocation;
+import com.powsybl.security.ViolationLocation;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.powsybl.security.json.SecurityAnalysisResultDeserializer.SOURCE_VERSION_ATTRIBUTE;
+
+/**
+ * @author Etienne Lesot {@literal <etienne.lesot at rte-france.com>}
+ */
+public class ViolationLocationDeserializer extends StdDeserializer<ViolationLocation> {
+
+    private static final String BUS_ID = "busId";
+    private static final String VOLTAGE_LEVEL_ID = "voltageLevelId";
+    private static final String BUS_BAR_IDS = "busbarIds";
+    public static final String SHORT_CIRCUIT_RESULT_VERSION_ATTRIBUTE = "shortCircuitResultVersion";
```

Comment:
Can you move this constant to `ShortCircuitAnalysisResultDeserializer`?
It would make it usable for other elements which could be introduced later.